docker/jenkins/Dockerfile.windows

```diff
@@ -1,15 +1,20 @@
 # escape=`
 
-# Use the latest Windows Server Core (LTSC - Long Term Service Channel) image with .NET Framework 4.8.
-# https://docs.microsoft.com/en-us/visualstudio/install/build-tools-container?view=vs-2019
+# https://hub.docker.com/r/microsoft/windows-servercore
 FROM mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2019
 
-# If building the docker image locally uncomment this line
-# ENV JENKINS_URL=something
-
-# Restore the default Windows shell for correct batch processing.
+# Use plain cmd shell for initial setup.
 SHELL ["cmd", "/S", "/C"]
 
+# Set JENKINS_URL if it's not currently set.
+RUN if not defined JENKINS_URL ( setx /M JENKINS_URL jenkins-docker-build )
```

Comment:
This should remove the need to set `JENKINS_URL` by hand. Does this seem like a sensible approach?